src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,160 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last, value, null);
+        if (first == null) {
+            first = newNode;
+        }
+        if (last != null) {
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkPositionIndex(index);
+        Node<T> currentNode = findNodeAtIndex(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        if (currentNode.prev == null) {
+            currentNode.prev = newNode;
+            first = newNode;
+        } else {
+            currentNode.prev.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> currentNode = findNodeAtIndex(index);
+        return currentNode.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> currentNode = findNodeAtIndex(index);
+        T oldValue = currentNode.value;
+        currentNode.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> findedNode = findNodeAtIndex(index);
+        removeNode(findedNode);
+        size--;
+        return findedNode.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> findedNode = findIndexAtValue(object);
+        if (findedNode != null) {
+            removeNode(findedNode);
+            size--;
+            return true;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> findNodeAtIndex(int index) {
```

Comment:
![Screenshot 2022-06-10 at 09 20 24](https://user-images.githubusercontent.com/60258373/173003181-e2ae5e83-802d-44e7-99f4-254c6196a85d.png)
Before you create PR with your homework, you need to go through the checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected